unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PluginSystem/PluginSystemFactory.cs

```diff
@@ -57,7 +58,7 @@ public static PluginSystem Create()
             pluginSystem.Register<FallbackFontsLoaderPlugin>(() => new FallbackFontsLoaderPlugin());
             pluginSystem.Register<SentryPlugin>(() => new SentryPlugin());
 
-            pluginSystem.RegisterWithFlag<LoadingScreenPlugin>(() => new LoadingScreenPlugin(), DataStore.i.featureFlags.DECOUPLED_LOADING_SCREEN_FF);
+            pluginSystem.RegisterWithFlag<LoadingScreenPlugin>(() => new LoadingScreenPlugin(Environment.i.serviceLocator.Get<IAddressableResourceProvider>()), DataStore.i.featureFlags.DECOUPLED_LOADING_SCREEN_FF);
```

Comment:
Same as before, you can use the `private Service<IAddressableResourceProvider> addresableResolver;` and the `.Ref` to access it

Reply:
I am not sure it make sense here. First of all because it is a static class with just one static method. Second, I don't want to hold a reference to it, but just pass it here.